app/submitter/convert_payload_0_0_1.py

```diff
@@ -107,14 +123,13 @@ def _get_checkbox_answer_data(answer_store, answer_schema, value):
         if option:
             if "detail_answer" in option:
                 detail_answer = answer_store.get_answer(option["detail_answer"]["id"])
-
                 # if the user has selected an option with a detail answer we need to find the detail answer value it refers to.
                 # the detail answer value can be empty, in this case we just use the main value (e.g. other)
-                user_answer = detail_answer.value or user_answer
+                user_answer = detail_answer.value or user_answer  # type: ignore
 
             qcodes_and_values.append((option.get("q_code"), user_answer))
 
-    checkbox_answer_data = OrderedDict()
+    checkbox_answer_data: dict = OrderedDict()
```

Comment:
Is this needed?. If so, it should match the return type.
```suggestion
    checkbox_answer_data: dict[str, str] = OrderedDict()
```